fastlane/lanes/release.rb

```diff
@@ -0,0 +1,236 @@
+# frozen_string_literal: true
+
+# Lanes related to the Release Process (Code Freeze, Betas, Final Build, AppStore Submission…)
+#
+platform :ios do
+  # Executes the initial steps needed during code freeze
+  #
+  # - Cuts a new release branch
+  # - Extracts the Release Notes
+  # - Freezes the GitHub milestone and enable GitHub branch protection for the new branch
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  #
+  desc 'Executes the initial steps needed during code freeze'
+  lane :code_freeze do |options|
+    gutenberg_dep_check
+    ios_codefreeze_prechecks(options)
+
+    ios_bump_version_release(skip_deliver: true)
+    new_version = ios_get_app_version
+
+    release_notes_source_path = File.join(PROJECT_ROOT_FOLDER, 'RELEASE-NOTES.txt')
+    extract_release_notes_for_version(
+      version: new_version,
+      release_notes_file_path: release_notes_source_path,
+      extracted_notes_file_path: extracted_release_notes_file_path(app: :wordpress)
+    )
+    # It would be good to update the action so that it can:
+    #
+    # - Use a custom commit message, so that we can differentiate between
+    #   WordPress and Jetpack
+    # - Have some sort of interactive mode, where the file is extracted and
+    #   shown to the user and they can either confirm and let the lane commit,
+    #   or modify it manually first and then run through the
+    #   show-confirm-commit cycle again
+    #
+    # In the meantime, we can make due with a duplicated commit message and the
+    # `print_release_notes_reminder` at the end of the lane to remember to make
+    # updates to the files.
+    extract_release_notes_for_version(
+      version: new_version,
+      release_notes_file_path: release_notes_source_path,
+      extracted_notes_file_path: extracted_release_notes_file_path(app: :jetpack)
+    )
+    ios_update_release_notes(new_version: new_version)
+
+    setbranchprotection(repository: GHHELPER_REPO, branch: "release/#{new_version}")
+    setfrozentag(repository: GHHELPER_REPO, milestone: new_version)
+    ios_check_beta_deps(podfile: File.join(PROJECT_ROOT_FOLDER, 'Podfile'))
+
+    print_release_notes_reminder
+  end
+
+  # Completes the final steps for the code freeze
```

Comment:
For consistency with the wording from the `code_freeze` lane:

```suggestion
  # Executes the final steps for the code freeze
```